resolver/resolver.go

```diff
@@ -285,7 +286,65 @@ func (resolver *Resolver) QueryStat(ctx context.Context, id interface{}, params
 }
 
 func (resolver *Resolver) QueryFees(ctx context.Context, id interface{}, params *jsonrpc.ParamsQueryFees, req *http.Request) jsonrpc.Response {
-	return resolver.handleMessage(ctx, id, jsonrpc.MethodQueryFees, *params, req, false)
+	// This is required for compatibility with renjs v1
+
+	reqWithResponder := lhttp.NewRequestWithResponder(ctx, id, jsonrpc.MethodQueryState, params, nil)
+	if ok := resolver.cacher.Send(reqWithResponder); !ok {
+		resolver.logger.Error("failed to send request to cacher, too much back pressure")
+		jsonErr := jsonrpc.NewError(jsonrpc.ErrorCodeInternal, "too much back pressure", nil)
+		return jsonrpc.NewResponse(id, nil, &jsonErr)
+	}
+
+	select {
+	case <-ctx.Done():
+		resolver.logger.Error("timeout when waiting for response: %v", ctx.Err())
+		jsonErr := jsonrpc.NewError(jsonrpc.ErrorCodeInternal, "request timed out", nil)
+		return jsonrpc.NewResponse(id, nil, &jsonErr)
+	case response := <-reqWithResponder.Responder:
+		raw, err := json.Marshal(response.Result)
+		if err != nil {
+			resolver.logger.Errorf("[resolver] error marshaling queryState result: %v", err)
+		}
+
+		var resp map[string]map[string]map[string]interface{}
+		if err := json.Unmarshal(raw, &resp); err != nil {
+			resolver.logger.Warnf("[resolver] cannot unmarshal queryState result from %v", err)
+		}
+
+		state := jsonrpc.ResponseQueryState{}
+		state.State = make(map[multichain.Chain]pack.Struct)
+
+		for i := range resp["state"] {
```

Comment:
Same as the unmarshal point but would prefer if we could do something like `resp.State`, `resp.State[i].GasCap`, etc.